File: eclipse_workspace/cmp_326_project_1/src/gameinterfaces/Room.java
package gameinterfaces;

/**
 * 
 * A class that implements this interface should have the following attributes:
 * <ul>
 * 
 * <li>roomIndex: int</li>
 * 
 * <li>description: String</li>
 * 
 * <li>monster: Monster</li>
 *
 * </ul>
 * 
 * The implementing class should only have the following constructor:
 * 
 * public Room(int roomIndex, String description, Monster monster)
 *
 */

public interface Room extends Player{

	/**
	 * This method returns <code>true</code> if the Monster in <code>this</code> Room has died.
	 * 
	 * @return <code>true</code> if   <code>this</code> Room <code>monster !isAlive()</code>, <code>false</code> otherwise.
	 * 
	 */
	public boolean isComplete();
	
	/**
	 * This method will display the message:
	 * <br>
	 * <br>
	 * "<code>player name</code>"
	 * "This Room is <code>this.description</code>, and has <code>this.monster</code>"
	 * <br>
	 * <br>
	 * If the monster <code>isAlive()</code> allow the player to enter the room and start a battle 
	 * @param player entering the room
	 */
	public void enter(Player player);

	/**
	 * 
	 * @return A String formatted as follows:
	 * <br>
	 * <br>
	 * "Room index %2d, description %20s, monster name %20s"
	 */
	@Override
	public String toString();
	


}

File: Qt_workspace/rpg_txt_game/room.cpp
#include "room.h"

Room::Room(){
    this->roomIndex = 0;
    this->description = "";
    this->monster = Monster();
}
Room::Room(int roomIndex, string description, Monster monster)
{
    this->roomIndex = roomIndex;
    this->description = description;
    this->monster = monster;
}

bool Room::isComplete(){
    return !monster.isAlive() ? true : false;
}

void Room::enter(Creature player){
    cout << player.getName() << "\nThis Room is " <<
             this->description << ", and has...\n" <<
            this->monster.getName() <<endl;
}

string Room::toString(){
    return "Room index: "+ to_string(this->roomIndex)+ ", Description: "
            + this->description + ", Monster name: " + monster.getName();
}


